pkg/apis/kops/validation/validation.go

```diff
@@ -1552,11 +1552,11 @@ func validateNodeLocalDNS(spec *kops.ClusterSpec, fldpath *field.Path) field.Err
 func validateClusterAutoscaler(cluster *kops.Cluster, spec *kops.ClusterAutoscalerConfig, fldPath *field.Path) (allErrs field.ErrorList) {
 	allErrs = append(allErrs, IsValidValue(fldPath.Child("expander"), spec.Expander, []string{"least-waste", "random", "most-pods", "price", "priority"})...)
 
-	if fi.StringValue(spec.Expander) == "price" && kops.CloudProviderID(cluster.Spec.CloudProvider) != kops.CloudProviderGCE {
+	if fi.StringValue(spec.Expander) == "price" && cluster.Spec.CloudProvider.GCE == nil {
```

Comment:
Should we just use `GetCloudProvider() != kops.CloudProviderGCE` ?  (Any reason for the different pattern?)

Reply:
I've been considering the getter to be largely transitional. I've been targeting the patterns we use with CNIs.

Reply:
Oh OK - I quite like the getter and I think it would be nice if we had a similar helper method for CNIs.  I don't think we need to surface it in the API, just that it seems to improve the code quality - less casting etc.  But just something to consider!